pkg/genericcomp/genericcomp.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package genericcomp
+
+import (
+	v1beta1v8o "github.com/verrazzano/verrazzano-crd-generator/pkg/apis/verrazzano/v1beta1"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	"github.com/verrazzano/verrazzano-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenericComponentSelectorLabel defines the selector label for generic component resources.
+const GenericComponentSelectorLabel = "verrazzano.name"
+
+// NewDeployment constructs a deployment for a generic component.
+func NewDeployment(generic v1beta1v8o.VerrazzanoGenericComponent, namespace string, labels map[string]string) *appsv1.Deployment {
```

Comment:
Does it make sense to have the Helidon component use this (not part of this effort)

Reply:
yes, we could do that and get rid of the helidon micro-operator